knotx-core/knotx-server/src/main/java/com/cognifide/knotx/server/KnotxServerVerticle.java

```diff
@@ -81,8 +81,8 @@ public void start(Future<Void> fut) throws IOException, URISyntaxException {
                               result -> {
                                 TemplateEngineResponse engineResponse = new TemplateEngineResponse(result.body());
                                 if (engineResponse.isSuccess()) {
-                                  rewriteHeaders(request, request.headers());
-                                  request.response().end(engineResponse.getHtml());
+                                  rewriteHeaders(request, repository.getHeaders());
+                                  request.response().setChunked(true).end(engineResponse.getHtml());
```

Comment:
as we discussed - chunked is to remove, and set header with content-length before response().end(...)